### PR TITLE
2537-handle-module-level-taxonomy-in-tests

### LINKED ISSUE #2537: Handle module level taxonomy in tests

### PR DESCRIPTION
**Pull request by clement-burtscher-sekoia**
## Description

Fixes https://github.com/SEKOIA-IO/intake-formats/issues/2537

- Update taxonomy tests to correctly handle module-level and format-level field scopes when checking missing and unused fields, to prevent false failures caused by module-level entries being asserted against format-level files
- Remove unused Beats fields to fix taxonomy tests:
  - Remove Beats/auditbeat unused field `ecs`
  - Remove Beats/winlogbeat unused field `event.origin`

## Type of Change

- [ ] New intake format
- [ ] Update to existing intake format
- [ ] Bug fix
- [ ] Documentation update
- [X] Other (please describe): fix tests

## Affected Intake Formats

- Vendor/product:

### 🔒 CRITICAL - Security Requirements (MUST BE VERIFIED FIRST)

- [ ] **No real email addresses** in test files (use: user@example.com, test@example.org)
- [ ] **No real passwords or API keys** (use: "P@ssw0rd!", "xxxxxxxxxxxx", "FAKE_TOKEN_123")
- [ ] **No real IP addresses** (use TEST-NET ranges: 192.0.2.x, 198.51.100.x, 203.0.113.x)
- [ ] **No real phone numbers** (use: +1-555-0100 or similar test numbers)
- [ ] **No real names or PII** (use: "John Doe", "Jane Smith", "User123")
- [ ] **No real company/organization names** in test data (unless public vendors)
- [ ] **No real usernames, hostnames, or domain names** (use: user1, host.example.com, test.corp)
- [ ] All test data has been anonymized and verified

### Required for All PRs

- [ ] Code has been linted
    - [ ] with the linter for manifest, smart-descriptions, and test files (`uv run python linter.py check --changes` in utils/)
    - [ ] with Prettier for the parser code files (`npx prettier --write <module>/<format>/ingest/parser.yml`)
- [ ] All CI/CD checks pass
- [ ] Changes follow the contribution guidelines in `CONTRIBUTING.md`

### Required for New/Updated Intake Formats

- [ ] Clear descriptions provided for modules and formats
- [ ] Logo files included for new modules/formats
- [ ] Parser test coverage is at least 75%
- [ ] At least one smart-description is provided
- [ ] README.md updated (if applicable)
- [ ] Taxonomy mappings are correct and documented
- [ ] Test cases cover edge cases and error handling

### Testing

- [ ] Local tests pass (`uv run pytest` in utils/)
- [ ] Sample data is representative and **completely anonymized**
- [ ] **No sensitive information** (PII, credentials, real IPs, real emails) in test files
- [ ] Test data uses example.com/example.org domains and TEST-NET IP ranges
- [ ] Parser handles malformed data gracefully

### Documentation

- [ ] Code changes are documented
- [ ] Smart-descriptions are clear and informative
- [ ] Field mappings are explained

## Related Issues

Closes #

## Summary by Sourcery

Adjust taxonomy validation tests to distinguish module-level and format-level field scopes, avoiding false positives when checking missing and unused fields.

Enhancements:
- Introduce utilities to split taxonomy fields by module and format scope for more accurate unused/missing field reporting.

Documentation:
- Add a utils changelog entry documenting the taxonomy test scope handling fix.

Tests:
- Update taxonomy-related tests to separately track and report module-level versus format-level missing and unused fields, while still pruning/fixing each scope appropriately.

### CODE REVIEW

File: utils/test_formats.py [hunk 1]
```diff
@@ -151,36 +209,57 @@ def prune_taxonomy(format_fields_path, taxonomy):
...
 
     # Remove each unused field from fields.yml
     print(request.config.getoption("prune_taxonomy"))
     if request.config.getoption("prune_taxonomy"):
-        prune_taxonomy(format_fields_path, taxonomy)
+        if number_of_unused_fields > 0:
+            prune_taxonomy(format_fields_path, {"unused": format_unused_fields})
+
+        if module_unused_fields:
+            prune_taxonomy(module_fields_path, {"unused": module_unused_fields})
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply utils/test_formats.py]: Fixed with commit [Apply sourcery-ai review suggestions](https://github.com/SEKOIA-IO/intake-formats/pull/2539/changes/525438bbe0bdc58110c6cf9a42b2b7cf3bcdc074)

File: utils/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+## [1.0.0] - 2026-05-21
+
+### Fixed
+
+- Update taxonomy tests to correctly handle module-level and format-level field scopes when checking missing and unused fields, to prevent false failures caused by module-level entries being asserted against format-level files
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply utils/CHANGELOG.md:14]: Fixed with commit [Align changelog on pyproject.toml](https://github.com/SEKOIA-IO/intake-formats/pull/2539/changes/6c205104ebcb145ba18df55a4f34020b6dcda26e)

**Review by mchupeau-sk — DISMISSED** [review]
For me it's not used, LGFM and it's not also a part of ECS field.

I'm agree with this solution

**Review by squioc — APPROVED** [review]
LGTM. Thank you for the work
